src/SFA.DAS.ApprenticeCommitments.Database/PostDeployment/AddMissingApprenticeIdsToRegistration.sql

```diff
@@ -0,0 +1,41 @@
+﻿/*
+Post-Deployment Script Template							
+--------------------------------------------------------------------------------------
+ This file contains SQL statements that will be appended to the build script.		
+ Use SQLCMD syntax to include a file in the post-deployment script.			
+ Example:      :r .\myfile.sql								
+ Use SQLCMD syntax to reference a variable in the post-deployment script.		
+ Example:      :setvar TableName MyTable							
+               SELECT * FROM [$(TableName)]					
+--------------------------------------------------------------------------------------
+*/
+
+-- This is only Registrations that are missing the ApprenticeshipId
+--select count(*) from registration where apprenticeshipid is null and commitmentsApprenticeshipId in (select commitmentsApprenticeshipId from revision)
+
+if exists(
+    select * from Registration
+    where ApprenticeshipId is null
+        and CommitmentsApprenticeshipId in (select CommitmentsApprenticeshipId from Revision))
+begin
+    print 'Adding missing Apprenticeship IDs to Registrations'
+
+    -- See if there are any Registrations that have more than one CommitmentsApprenticeshipId in a Revision
+    -- THIS SHOULD RETURN ZERO RESULTS
+    if exists (
+        select top 1000 RegistrationId, count(distinct Revision.ApprenticeshipId) NumApprs
+        from Registration inner join Revision on Registration.CommitmentsApprenticeshipId = Revision.CommitmentsApprenticeshipId
+        group by RegistrationId
+        having count(distinct Revision.ApprenticeshipId) > 1
+    )
+    begin
+        raiserror (N'There are multiple apprentices for a registration', 16, 127) with nowait
```

Comment:
Now I always get a little confused about how this SQL exception works. if the level is 20 or above, the connection is terminated (not follow statements are executed, but below 20 I think the subsequent code is executed. 

But because the exception is raised it would (i think) be rolled back. It might be worth putting the update inside an else block

Reply:
It seemed to work when I tested it with multiple registrations in the db, but it makes sense to add extra protection by using `if` 👍